examples/advanced/lr-newton-raphson/job/newton_raphson/app/custom/newton_raphson_train.py

```diff
@@ -0,0 +1,229 @@
+
+def validate(data, theta):
+  """
+  Performs local validation.
+  Computes accuracy and precision scores.
+
+  """
+  valid_X = data['valid_X']
+  valid_y = data['valid_y']
+
+  # Add intercept, pre-pend 1s to as first
+  # column of valid_X
+  valid_X = np.concatenate(
+    (np.ones((valid_X.shape[0], 1)), valid_X),
+    axis=1
+  )
+
+  # Compute probabilities from current weights
+  proba = sigmoid(np.dot(valid_X, theta))
+
+  return {
+    'accuracy': accuracy_score(valid_y, proba.round()),
+    'precision': precision_score(valid_y, proba.round())
+  }
+
+
+def main():
+  """
+  This is a typical ML training loop,
+  augmented with Flare Client API to
+  perform local training on each client
+  side and send result to server.
+
+  """
+  args = parse_arguments()
+
+  flare.init()
+
+  site_name = flare.get_site_name()
+  print('training on client site: {}'.format(site_name))
+
+  # Load client site data.
+  data = load_data(
+    args.data_root,
+    site_name
+  )
+
+  # Get metric summary writer
+  writer = SummaryWriter()
+
+  while flare.is_running():
+
+    # Receive global model (FLModel) from server.
+    global_model = flare.receive()
+
+    curr_round = global_model.current_round
+    print('current_round={}'.format(curr_round))
+
+    print(('[ROUND {}] - client site: {}, received '
+           'global model: {}').format(
+             curr_round, site_name, global_model))
+
+    # Get the weights, aka parameter theta for
+    # logistic regression.
+    global_weights = global_model.params[NPConstants.NUMPY_KEY]
+    print('[ROUND {}] - global model weights: {}'.format(
+      curr_round, global_weights))
+
+    # Local validation before training
+    print(('[ROUND {}] - start validation of global '
+           'model on client: {}').format(
+             curr_round, site_name))
+    validation_scores = validate(data, global_weights)
+    print(('[ROUND {}] - validation metric scores on '
+           'client: {} = {}').format(
+             curr_round, site_name, validation_scores))
+
+    # Write validation metric summary
+    writer.add_scalar(
+      '{}/accuracy'.format(site_name),
+      validation_scores['accuracy'],
+      curr_round
+    )
+
+    writer.add_scalar(
+      '{}/precision'.format(site_name),
+      validation_scores['precision'],
+      curr_round
+    )
+
+    # Local training
+    print(('[ROUND {}] - start local training on client '
+           'site: {}').format(
+             curr_round, site_name))
+    result_dict = train_newton_raphson(
+      data,
+      theta=global_weights
+    )
+
+    # Send result to server for aggregation.
+    result_model = FLModel(
+      params=result_dict,
+      params_type=ParamsType.FULL
+    )
+    result_model.meta[FLMetaKey.NUM_STEPS_CURRENT_ROUND] = data['train_X'].shape[0]
+
+    print(
+      ('[ROUND {}] - local newton raphson training from '
+       'client: {} complete, sending results to server: {}').format(
+         curr_round, site_name, result_model))
```

Comment:
Do you want to print the entire FLModel here? Do we have a way to print a summary of the model instead? @chesterxgchen @YuanTingHsieh

Reply:
I mean summary like the ExchangeObject in MONAI FL is doing https://github.com/Project-MONAI/MONAI/blob/e1a69b03c86ce065db2816b696ea4a6b57d46435/monai/fl/utils/exchange_object.py#L103

Reply:
It only shows the length of the params [dict](https://github.com/Project-MONAI/MONAI/blob/e1a69b03c86ce065db2816b696ea4a6b57d46435/monai/fl/utils/exchange_object.py#L88) or type if not a dict. Hence it shows a "summary" of the params that will be sent.

Reply:
I filed a new [issue](https://github.com/NVIDIA/NVFlare/issues/2538) for this request. @zhijinl , you can leave your code as is if we address the issue.